Mathlib/Data/Real/EReal.lean

```diff
@@ -900,6 +900,24 @@ theorem add_lt_top {x y : EReal} (hx : x ≠ ⊤) (hy : y ≠ ⊤) : x + y < ⊤
   exact EReal.add_lt_add hx.lt_top hy.lt_top
 #align ereal.add_lt_top EReal.add_lt_top
 
+theorem add_le_add_right {x y z : EReal} (h : x ≤ y) : x + z ≤ y + z := by
```

Comment:
Instead of registering a specific lemma for EReal, can you instead register the right typeclass so that the general lemma `add_le_add_right` applies? Same thing for the other lemmas below.

Reply:
We have just removed our EReal-specialised implementation of `add_le_add_right` and `add_le_add_left` since we have noticed that the more general `_root_. add_le_add_right` and `_root_. add_le_add_left` worked perfectly well. 

We were just applying them incorrectly...